cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -378,6 +382,23 @@ public void updateApp(FullHttpRequest request, HttpResponder responder,
     }
   }
 
+  /**
+   * Get a list of {@link ApplicationDetail} for all applications in the given namespace
+   *
+   * @param request   {@link HttpRequest}
+   * @param responder {@link HttpResponse}
+   * @param namespace the namespace to get all application details
+   * @throws Exception if namespace doesn't exists or failed to get all application details
+   */
+  @GET
+  @Path("/appdetail")
```

Comment:
what's the reason the other app list endpoint was changed? Are we causing issues by adding back the old functionality?

Reply:
The inconsistency between /apps returning in ApplicationDetails format while /app/<appid> etc returning ApplicationRecord has been there for quite a while (since the acquisition from what I can tell from the change history). 

Unable to see change history past the acquisition (since it was in a different repo?)

Anyway, regardless of why the inconsistency was introduced, this inconsistency looks ugly and error prone. I didn't notice this and my unit test didn't catch this due to json deserialization just ignore unknown fields, so unit tests on the client side needs to check as many fields as possible to ensure this gets caught. 

I would prefer not to use these endpoints going forward in hope of deprecating them somehow (probably not possible), but at least not adding new code use them.

Reply:
I misunderstood the commit comments, so there isn't any change. It's not an uncommon pattern for a list to return less information than a get on the specific element returns.

Should this be an internal handler then instead of mixed in with the user facing APIs?

Reply:
Ack. 

Yes. Moved these to internal. Keep public unchanged.